srcpkgs/docker-cli/template

```diff
@@ -0,0 +1,39 @@
+# Template file for 'docker-cli'
+pkgname=docker-cli
+version=20.10.5
+revision=1
+archs="aarch64* armv[67]l* x86_64* ppc64le*"
+create_wrksrc=yes
+build_style=go
+go_import_path="github.com/docker/cli"
+hostmakedepends="pkg-config tar"
+depends="containerd runc iptables xz git"
+short_desc="Pack, ship and run any application as a lightweight container - cli"
+maintainer="PWA COLLECTIVE <mail@pwa.cya.cx>"
+license="Apache-2.0"
+homepage="http://www.docker.io"
+distfiles="https://github.com/docker/cli/archive/v${version}.tar.gz"
+checksum=4ba845f8c7e2e0a2ca1ec6589847159ca8d0d37b609f0e6f78def7a893b9b342
+system_groups="docker"
+
+_cli_commit=55c4c88966a912ddb365e2d73a4969e700fc458f
+_cli_builddir="cli-$version"
+
+export GO111MODULE=off # go1.16 defaults to on
+
+do_build() {
+	export AUTO_GOPATH=1
+	export GITCOMMIT=$_cli_commit
+	export DISABLE_WARN_OUTSIDE_CONTAINER=1
+
+	mkdir -p "$wrksrc/_cli_builddir"
+
+	cd "$wrksrc/$_cli_builddir"
+	mkdir -p src/github.com/docker/
+	ln -sf "$wrksrc/$_cli_builddir" src/github.com/docker/cli
+	GOPATH="$PWD" LDFLAGS="" VERSION="$version" make dynbinary
+}
```

Comment:
The original `docker` template had to build multiple components in its custom `do_build`. Have you tried using the stanard `do_build` from the `go` build style? All the `make dynbinary` target does is call a script which sources an environment and then calls `go build`. You might be able just to the necessary variables directly here and use the standard behavior.

Reply:
I attempted removing the custom do_build() and do_install() functions, however builds fail with the error:

```
=> ERROR: docker-cli-20.10.5_1: do_build: 'go get -p "$XBPS_MAKEJOBS" -x -tags "${go_build_tags}" -ldflags "${go_ldflags}" ${go_package}' exited with 1
=> ERROR:   in do_build() at common/build-style/go.sh:39
```

when this is set. I don't know how to fix this, so I'm going to leave the custom functions in, as they compile just fine and I don't see how they could impact build performance if they are similar to the default. Feel free to tell me what the issue is here, as I don't have enough experience with the xbps build system to understand it at 2am.

Reply:
I'll take a look to see if there is an easy fix, but I'm not too worried about it if making the standard function work requires a massive effort.

Reply:
This is resolved with the latest commits.